### PR TITLE
feat: Add option to replace values in remote url

### PR DESCRIPTION
**Pull request by abeatrix**
This is a PR to add configure option `remoteUrlReplacements` used in [our VS Code plugin](https://github.com/sourcegraph/sourcegraph-vscode) which allows users to replace values in remote url.

Users can enable this feature by adding `remoteUrlReplacements` to their global configuration file `sourcegraph-jetbrains.properties` or repository-based configuration file `.idea/sourcegraph.xml`. 

This feature will not be enabled if less than two or more than four values were passed into the configuration.

## Examples

image
image
URL changed from https://github.com/sourcegraph/sourcegraph-jetbrains.git to https://gitexample.com/s/s-jetbrains.git using the configurations shown above
image

This feature should address the request listed in https://github.com/sourcegraph/sourcegraph-jetbrains/issues/17

### CODE REVIEW

**Review by coury-clark — COMMENTED** [review]
The general approach is good, there is just a small issue where not all of the actions are getting this behavior. I gave some thoughts in the comments for how you might address that.

File: src/main/java/FileAction.java [hunk 1]
```diff
@@ -54,18 +56,29 @@ public void actionPerformed(AnActionEvent e) {
...
+        // set defaultBranch if config option is not null
+        String branch = Util.setDefaultBranch(project)!=null ? Util.setDefaultBranch(project) : repoInfo.branch;
+        String remoteURL = repoInfo.remoteURL;
 
-        // set defaultBranch only if not config is not null
-        if(Util.setDefaultBranch(project)!=null) {
-            branch = Util.setDefaultBranch(project);
-        };
+        // replace remoteURL if config option is not null
+        if(Util.setRemoteUrlReplacements(project)!=null) {
```
[the remark this answers is not in the document]
Comment (abeatrix) [reply src/main/java/FileAction.java]: Thanks for the suggestion Coury! I have moved this logic to Util.repoInfo to reflect this!

File: src/main/java/FileAction.java [hunk 2]
```diff
@@ -54,18 +56,29 @@ public void actionPerformed(AnActionEvent e) {
...
+            String r = Util.setRemoteUrlReplacements(project);
+            String[] replacements = r.trim().split("\\s*,\\s*");
+            // Check if the entered values are pairs with no more than 2 pairs
```
[the remark this answers is not in the document]
Comment (abeatrix) [reply src/main/java/FileAction.java]: 🤷‍♀️

File: src/main/java/SearchActionBase.java [hunk 3]
```diff
@@ -51,12 +51,23 @@ public void actionPerformedMode(AnActionEvent e, String mode) {
...
+        // set defaultBranch if config option is not null
+        String branch = Util.setDefaultBranch(project)!=null ? Util.setDefaultBranch(project) : repoInfo.branch;
+        String remoteURL = repoInfo.remoteURL;
 
-        // set defaultBranch only if not config is not null
-        if(Util.setDefaultBranch(project)!=null) {
-            branch = Util.setDefaultBranch(project);
-        };
+        // replace remoteURL if config option is not null
+        if(Util.setRemoteUrlReplacements(project)!=null) {
```
[the remark this answers is not in the document]
Comment (abeatrix) [reply src/main/java/SearchActionBase.java]: Moved this logic to Util.repoInfo to reflect this as well :)

**Review by muratsu — APPROVED** [review]

**Pull request merged**